opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java

```diff
@@ -24,54 +25,74 @@
  */
 @RequiredArgsConstructor
 public abstract class RelevanceQuery<T extends QueryBuilder> extends LuceneQuery {
+  @Getter
   private final Map<String, QueryBuilderStep<T>> queryBuildActions;
 
-  @Override
-  public QueryBuilder build(FunctionExpression func) {
-    var arguments = func.getArguments().stream()
-        .map(a -> (NamedArgumentExpression)a).collect(Collectors.toList());
-    if (arguments.size() < 2) {
-      throw new SyntaxCheckException(
-          String.format("%s requires at least two parameters", getQueryName()));
+  protected void ignoreArguments(List<NamedArgumentExpression> arguments) {
+    arguments.removeIf(a -> a.getArgName().equalsIgnoreCase("field")
+            || a.getArgName().equalsIgnoreCase("fields")
+            || a.getArgName().equalsIgnoreCase("query"));
+  }
+
+  protected void checkValidArguments(String argNormalized, T queryBuilder) {
+    if (!queryBuildActions.containsKey(argNormalized)) {
+      throw new SemanticCheckException(
+              String.format("Parameter %s is invalid for %s function.",
+                      argNormalized, queryBuilder.getWriteableName()));
     }
+  }
 
+  protected T loadArguments(List<NamedArgumentExpression> arguments) throws SemanticCheckException {
     // Aggregate parameters by name, so getting a Map<Name:String, List>
     arguments.stream().collect(Collectors.groupingBy(a -> a.getArgName().toLowerCase()))
-        .forEach((k, v) -> {
-          if (v.size() > 1) {
-            throw new SemanticCheckException(
-                String.format("Parameter '%s' can only be specified once.", k));
-          }
-        });
+            .forEach((k, v) -> {
```

Comment:
Formatting change not needed.

Reply:
Introduced by accident when I was trying to fix checkstyle issues. Fixed in 7add05fe7c1a7fcef259e1e10656f3c09e50bbc8